x-pack/test/defend_workflows_cypress/runner.ts

```diff
@@ -9,7 +9,44 @@ import { resolve } from 'path';
 import Url from 'url';
 import { withProcRunner } from '@kbn/dev-proc-runner';
 
+import { startRuntimeServices } from '@kbn/security-solution-plugin/scripts/endpoint/endpoint_agent_runner/runtime';
 import { FtrProviderContext } from './ftr_provider_context';
+import { AgentManager } from './agent';
+import { FleetManager } from './fleet_server';
+
+async function withFleetAgent(
+  { getService }: FtrProviderContext,
+  runner: (runnerEnv: Record<string, string>) => Promise<void>
+) {
+  const log = getService('log');
+  const config = getService('config');
+
+  const elasticUrl = Url.format(config.get('servers.elasticsearch'));
+  const kibanaUrl = Url.format(config.get('servers.kibana'));
+  const username = config.get('servers.elasticsearch.username');
+  const password = config.get('servers.elasticsearch.password');
+
+  await startRuntimeServices({
+    log,
+    elasticUrl,
+    kibanaUrl,
+    username,
+    password,
+    version: '8.7.0-SNAPSHOT',
```

Comment:
should we use `getAgentVersionMatchingCurrentStack()` from `x-pack/plugins/security_solution/scripts/endpoint/common/fleet_services.ts` so that we don't have to always be updating this value with every release?

Actually: I don't even know if we should use `-SNAPSHOT` for these tests. The Agent/Endpoint (and maybe even fleet server) are not always in the best stable condition during the development cycle and that could really turns these tests to be flaky. 

I also found (last week) that when the release branch is cut, which means the `package.json` version is updated to the next release, no `SNAPSHOT`'s exist, so tests would also fail then. 

Maybe we use the prior minor to the current stack version? `semver` module might help determine that value by using the current stack version and then moving the `minor` number back by `-1`.

Reply:
https://github.com/elastic/kibana/pull/148893/files#diff-5eb7821f65370aa32578be8c9d7fa2b99b4d2624422a398703b8c7f78f2702d4R228-R236
I've added a check to always install the latest matching agent unless the version doesn't exist on the artifacts API anymore, in that case, it will use the version of the stack

I agree that Agent is not always the most stable during the cycle, but I believe we should be using `-SNAPSHOT` versions, otherwise, we won't be able to catch the bugs/test new functionality properly

Please let me know what you think

Reply:
Remind me again: will this tst suite be enable in CI by default? if so, then IMO its going to be too unstable - but we can try. 

if the goal is to use this test suite for manual testing, then I'm ok with you keeping `-SNAPSHOT`.